src/lgdo/lh5_store.py

```diff
@@ -773,12 +768,25 @@ def read_object(
             if n_rows_to_read > n_rows:
                 n_rows_to_read = n_rows
 
+            # if idx is passed, check if we can make it a slice instead (faster)
+            change_idx_to_slice = False
+
             # prepare the selection for the read. Use idx if available
             if idx is not None:
-                source_sel = idx
+                # check if idx is empty and convert to slice instead
+                if len(idx[0]) == 0:
+                    source_sel = np.s_[0 : 0]
+                    change_idx_to_slice = True
+                # check if idx is contiguous and increasing
+                # if so, convert it to a slice instead (faster)
+                elif np.all(np.diff(idx[0]) == 1):
```

Comment:
I think slices with steps != 1 are supported as well.

Reply:
I changed it to allow any consistent step size but this is now slower - I checked with step size of 2. My guess is that h5py doesn't support this well since it is still not accessing contiguous rows? It is faster to use the fancy indexing in this case. So I will not commit this change.

![test_slicestepof2](https://github.com/legend-exp/legend-pydataobj/assets/139183994/c73ad41a-a09d-4c81-a032-bc89cc8ef25e)


```python
            # if idx is passed, check if we can make it a slice instead (faster)
            change_idx_to_slice = False

            # prepare the selection for the read. Use idx if available
            if idx is not None:
                # check if idx is empty and convert to slice instead
                if len(idx[0]) == 0:
                    source_sel = np.s_[0:0]
                    change_idx_to_slice = True
                # check if idx has regular steps
                # if so, convert it to a slice instead (faster)
                else:
                    idxsteps = np.diff(idx[0])
                    if np.all(idxsteps == idxsteps[0]):
                        source_sel = np.s_[idx[0][0] : idx[0][-1] + 1 : idxsteps[0]]
                        change_idx_to_slice = True
                
                if not change_idx_to_slice:
                    source_sel = idx
            else:
                source_sel = np.s_[start_row : start_row + n_rows_to_read]
```

Reply:
OK, I guess it makes sense that HDF5 only supports continguous slices.